### PR TITLE
fix(controller): Adds ALL_POD_CHANGES_SIGNIFICANT

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/controller/pod/significant.go [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 import (
+	"os"
+
 	apiv1 "k8s.io/api/core/v1"
 )
 
 func SignificantPodChange(from *apiv1.Pod, to *apiv1.Pod) bool {
-	return from.Spec.NodeName != to.Spec.NodeName ||
+	return os.Getenv("ALL_POD_CHANGES_SIGNIFICANT") == "true" ||
```
Comment (simster7) [inline workflow/controller/pod/significant.go:10]: Flag is off by default? Would we want it on by default?

Comment (alexec) [reply workflow/controller/pod/significant.go:10]: this is actually covered by a test, but good to ask

**Comment by alexec** [comment]
This is available in https://github.com/argoproj/argo/releases/tag/v2.10.0-rc6

**Comment by alexec** [comment]
Available in [v2.9.5](https://github.com/argoproj/argo/releases/tag/v2.9.5)

**Pull request merged**